core/src/main/java/org/mule/TransformationService.java

```diff
@@ -48,36 +48,35 @@ public TransformationService(MuleContext muleContext)
     }
 
     /**
-     * Applies a list of transformers to the payload of the message. This *Will* change the payload of the
-     * message.
+     * Applies a list of transformers returning the result of the transformation as a new message instance.
```

Comment:
Regarding the "as a new message instance" part. Is a requirement that this method always return a new instance? What if the requested transformation is not required because the message payload is already an instance of the output type?